Client/app/src/main/java/ru/hse/client/SignInActivity.kt

```diff
@@ -0,0 +1,215 @@
+package ru.hse.client
+
+
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.widget.Button
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+import okhttp3.*
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okio.ByteString
+import ru.hse.server.proto.EntitiesProto
+import java.io.IOException
+
+
+class SignInActivity : AppCompatActivity() {
```

Comment:
This is a fairly complex class that includes a lot of actions. Think about how you can simplify it